ocs_ci/utility/utils.py

```diff
@@ -2928,6 +2928,35 @@ def convert_device_size(unformatted_size, units_to_covert_to):
     return conversion[units_to_covert_to][units]
 
 
+def convert_bytes_to_unit(bytes_to_convert):
```

Comment:
this function cant convert from [something] to bytes. i did try that function first

Reply:
With a small change it can. It doesn't work but it doesn't mean we need to write new one.
This will fix it. Please try.
`    byte_pattern = re.compile("[^A-Z]*B$")
    if byte_pattern.match(unformatted_size):
        units = unformatted_size[-1:]
        abso = int(unformatted_size[:-1])
    else:
        units = unformatted_size[-2:]
        abso = int(unformatted_size[:-2])
    conversion = {
        "TB": {"Ti": abso, "Gi": abso / 1000, "Mi": abso / 1e6, "Ki": abso / 1e9, "B": abso / 1e10},
        "GB": {"Ti": abso * 1000, "Gi": abso, "Mi": abso / 1000, "Ki": abso / 1e6, "B": abso / 1e7},
        "MB": {"Ti": abso * 1e6, "Gi": abso * 1000, "Mi": abso, "Ki": abso / 1000, "B": abso / 10000},
        "KB": {"Ti": abso * 1e9, "Gi": abso * 1e6, "Mi": abso * 1000, "Ki": abso, "B": abso / 1000},
        "B": {"Ti": abso * 1e12, "Gi": abso * 1e9, "Mi": abso * 1e6, "Ki": abso * 1000, "B": abso},
    }
    return conversion[units_to_covert_to][units]
`

Reply:
Please fix convert_device_size with this code.
diff --git a/ocs_ci/utility/utils.py b/ocs_ci/utility/utils.py
index e96935b1..93914846 100644
--- a/ocs_ci/utility/utils.py
+++ b/ocs_ci/utility/utils.py
@@ -2916,14 +2916,19 @@ def convert_device_size(unformatted_size, units_to_covert_to):
         int: The converted size
 
     """
-    units = unformatted_size[-2:]
-    abso = int(unformatted_size[:-2])
+    byte_pattern = re.compile("[^A-Z]*B$")
+    if byte_pattern.match(unformatted_size):
+        units = unformatted_size[-1:]
+        abso = int(unformatted_size[:-1])
+    else:
+        units = unformatted_size[-2:]
+        abso = int(unformatted_size[:-2])
     conversion = {
-        "TB": {"Ti": abso, "Gi": abso / 1000, "Mi": abso / 1e6, "Ki": abso / 1e9},
-        "GB": {"Ti": abso * 1000, "Gi": abso, "Mi": abso / 1000, "Ki": abso / 1e6},
-        "MB": {"Ti": abso * 1e6, "Gi": abso * 1000, "Mi": abso, "Ki": abso / 1000},
-        "KB": {"Ti": abso * 1e9, "Gi": abso * 1e6, "Mi": abso * 1000, "Ki": abso},
-        "B": {"Ti": abso * 1e12, "Gi": abso * 1e9, "Mi": abso * 1e6, "Ki": abso * 1000},
+        "TB": {"Ti": abso, "Gi": abso / 1000, "Mi": abso / 1e6, "Ki": abso / 1e9, "B": abso / 1e10},
+        "GB": {"Ti": abso * 1000, "Gi": abso, "Mi": abso / 1000, "Ki": abso / 1e6, "B": abso / 1e7},
+        "MB": {"Ti": abso * 1e6, "Gi": abso * 1000, "Mi": abso, "Ki": abso / 1000, "B": abso / 10000},
+        "KB": {"Ti": abso * 1e9, "Gi": abso * 1e6, "Mi": abso * 1000, "Ki": abso, "B": abso / 1000},
+        "B": {"Ti": abso * 1e12, "Gi": abso * 1e9, "Mi": abso * 1e6, "Ki": abso * 1000, "B": abso},
     }
     return conversion[units_to_covert_to][units]

Reply:
If something not working we need to fix it. They didn't consider that B is one letter and they are doing [-2] to get the suffix.
Please fix the function with this code and not write a new one. I know you invested but there is no need for 2 functions doing the same.
    byte_pattern = re.compile("[^A-Z]*B$")
    if byte_pattern.match(unformatted_size):
        units = unformatted_size[-1:]
        abso = int(unformatted_size[:-1])
    else:
        units = unformatted_size[-2:]
        abso = int(unformatted_size[:-2])
    conversion = {
        "TB": {"Ti": abso, "Gi": abso / 1000, "Mi": abso / 1e6, "Ki": abso / 1e9, "B": abso / 1e10},
        "GB": {"Ti": abso * 1000, "Gi": abso, "Mi": abso / 1000, "Ki": abso / 1e6, "B": abso / 1e7},
        "MB": {"Ti": abso * 1e6, "Gi": abso * 1000, "Mi": abso, "Ki": abso / 1000, "B": abso / 10000},
        "KB": {"Ti": abso * 1e9, "Gi": abso * 1e6, "Mi": abso * 1000, "Ki": abso, "B": abso / 1000},
        "B": {"Ti": abso * 1e12, "Gi": abso * 1e9, "Mi": abso * 1e6, "Ki": abso * 1000, "B": abso},
    }
    return conversion[units_to_covert_to][units]

Reply:
Also added B to conversion

Reply:
You function is using 1024 which is the right size, we need also to have to option to convert with 1024 so I suggest:
`def convert_device_size(unformatted_size, units_to_covert_to, convert=1000):
    """
    Convert a string representing a size to an int according to the given units
    to convert to

    Args:
        unformatted_size (str): The size to convert (i.e, '1Gi'/'100Mi')
        units_to_covert_to (str): The units to convert the size to (i.e, TB/GB/MB)
        convert (int): some convertiion need 1000, some 1024. default 1000
    Returns:
        int: The converted size

    """
    byte_pattern = re.compile("[^A-Z]*B$")
    if byte_pattern.match(unformatted_size):
        units = unformatted_size[-1:]
        abso = int(unformatted_size[:-1])
    else:
        units = unformatted_size[-2:]
        abso = int(unformatted_size[:-2])
    if convert == 1000:
        conversion = {
            "TB": {"Ti": abso, "Gi": abso / 1000, "Mi": abso / 1e6, "Ki": abso / 1e9, "B": abso / 1e10},
            "GB": {"Ti": abso * 1000, "Gi": abso, "Mi": abso / 1000, "Ki": abso / 1e6, "B": abso / 1e7},
            "MB": {"Ti": abso * 1e6, "Gi": abso * 1000, "Mi": abso, "Ki": abso / 1000, "B": abso / 10000},
            "KB": {"Ti": abso * 1e9, "Gi": abso * 1e6, "Mi": abso * 1000, "Ki": abso, "B": abso / 1000},
            "B": {"Ti": abso * 1e12, "Gi": abso * 1e9, "Mi": abso * 1e6, "Ki": abso * 1000, "B": abso},
        }
    elif convert == 1024:
        conversion = {
            "TB": {"Ti": abso, "Gi": abso / 1024, "Mi": abso / 1024e3, "Ki": abso / 1024e6, "B": abso / 1024e7},
            "GB": {"Ti": abso * 1024, "Gi": abso, "Mi": abso / 1024, "Ki": abso / 1024e3, "B": abso / 1024e4},
            "MB": {"Ti": abso * 1024e3, "Gi": abso * 1024, "Mi": abso, "Ki": abso / 1024, "B": abso / 10240},
            "KB": {"Ti": abso * 1024e6, "Gi": abso * 1024e3, "Mi": abso * 1024, "Ki": abso, "B": abso / 1024},
            "B": {"Ti": abso * 1024e9, "Gi": abso * 1024e6, "Mi": abso * 1024e3, "Ki": abso * 1024, "B": abso},
        }`
And send it with convert=1024

Reply:
this function working fine for other use cases, i dont want to make change the might break other tests in this PR. i will open an issue for that.

Reply:
I don't think we should be afraid to change an already existing functions. For this we have PR validation for tier1.
If you still wan't to have your own function so you will not send any unit (M,G etc) please have a look of this implementation https://github.com/shyRozen/ocs-ci/blob/638dbdebeb22e20a38831dbf0fc665c24873ba5c/ocs_ci/utility/utils.py#L2657-L2674 . The code is not merged, just I think it's more clean. Need just to remove the space from the return.
Anyway I let you decide how you wish to progress. Will merge with this implementation also.

Reply:
I agree with Shay. The existing function is very similar

Reply:
There is an issue on this function 
units = unformatted_size[-2:] and len("B")=1